tempesta_fw/http.c

```diff
@@ -2930,6 +2959,26 @@ tfw_http_req_process(TfwConn *conn, const TfwFsmData *data)
 		skb = NULL;
 	}
 
+	s_ip = tfw_http_get_ip_from_xff(req);
+	if (!TFW_STR_EMPTY(&s_ip)) {
+		TfwClient *cli, *conn_cli;
+		TfwStr *ua;
+
+		if (tfw_addr_pton(&s_ip, &addr) != 0)
+			return TFW_BLOCK;
+
+		conn_cli = (TfwClient *)conn->peer;
+		ua = &req->h_tbl->tbl[TFW_HTTP_HDR_USER_AGENT];
+		tfw_http_msg_clnthdr_val(ua, TFW_HTTP_HDR_USER_AGENT,
+					 &s_user_agent);
+		cli = tfw_client_obtain(conn_cli->addr, &addr, &s_user_agent,
+					NULL);
+		if (cli && cli != conn_cli)
+			req->peer = cli;
+		else
+			tfw_client_put(cli);
+	}
+
```

Comment:
Why the whole this block can't be moved to a new function like `tfw_http_req_client_link()` and here we can call only
```C
if ((r = tfw_http_req_cient_link(req)))
    return r;
```